File: cs/sample/Sample/SameTree/SameTree.cs
using System;
using System.Collections.Generic;
using Sample.TreeNode;

namespace Sample.SameTree{
    public class sameTree{
        public static bool naive(treeNode p,treeNode q){
            if (p==null&&q==null) return true;
            if (p==null||q==null) return false;
            if (p.val!=q.val) return false;
            return sameTree.naive(p.left,q.left)&&sameTree.naive(p.right,q.right);
        }
    }
}

File: cpp/example/src/SameTree/sameTree.cpp
#include "sameTree.h"

bool sameTree::naive(TreeNode* a,TreeNode* b){
    if (a==nullptr && b==nullptr) return true;
    if (a==nullptr || b==nullptr) return false;
    if (a->val!=b->val) return false;
    return sameTree::naive(a->left,b->left) && sameTree::naive(a->right,b->right);
}
